### ISSUE TITLE
Source Link opened in the same tab

### ISSUE THREAD

**Issue by llmwesee** [h2oai/h2ogpt#1678]
After got the answer and go to the source and hit the source link then it opened in the same tab instead of new tab. By looking the code, The link should open in the new tab always instead of same tab because we are targeting the blank. Then is it a Bug or something else?

```
def get_url(x, from_str=False, short_name=False, font_size=2):
    if not from_str:
        source = x.metadata['source']
    else:
        source = x
    if short_name:
        source_name = get_short_name(source)
    else:
        source_name = source
    if source.startswith('http://') or source.startswith('https://'):
        return """<font size="%s"><a href="%s" target="_blank"  rel="noopener noreferrer">%s</a></font>""" % (
            font_size, source, source_name)
    elif '<a href=' not in source:
        return """<font size="%s"><a href="file/%s" target="_blank"  rel="noopener noreferrer">%s</a></font>""" % (
            font_size, source, source_name)
    else:
        # already filled
        return source
```

**Comment by pseudotensor** [comment]
Sorry for delay.

I've never seen it open in same tab.  For me a PDF link leads to downloading of the PDF.

For images, it opens in new tab always for me.

**Comment by llmwesee** [comment]
I'm attaching the both PDF and also the screenshot for the reference. 

[Screencast from 28-06-24 09:45:42 AM IST.webm](https://github.com/h2oai/h2ogpt/assets/137979399/256c5ded-3f4d-4f68-a2dd-ef19b7cd1d31)

[2210.14699v2.pdf](https://github.com/user-attachments/files/16024081/2210.14699v2.pdf)

**Comment by pseudotensor** [comment]
I see.  It must be because for me it downloads the PDF in Ubuntu, but for you it is instead opening the PDF up.  And you are using some modified code to get the correct page number, which is great.

**Comment by llmwesee** [comment]
Yep, I want to open the PDF up. And for get the correct page number it doesn't work on cases where PDFs are the scanned copies.

**Comment by Sivakajan-tech** [comment]
I would like to contribute to this issue. 

Windows often requires file paths to be formatted differently. The path used is formatted correctly for Window. Using file:/// for better cross-platform support:

```
return """<font size="%s"><a href="file:///%s" target="_blank"  rel="noopener noreferrer">%s</a></font>""" % (
```

Note the three slashes (file:///) in the link. This makes the link an absolute path which can be more reliable across different platforms.

Ref: https://stackoverflow.com/questions/25354048/difference-between-file-and-file
